### ISSUE TITLE
Downloading Big Text Files (with long lines) Corrupts Them

### ISSUE THREAD

**Issue by gregarios** [proftpd/proftpd#696]
### What I Did

Uploaded a text file in ASCII mode using MacOS Fetch 5.7.7.
The file is about 80K, and has very long lines because it is a series of javascripts minified and stored in one big file. Download the same file and then it has a small group of strange unknown characters in it, replacing some of the text in the javascript code.

### What I Expected/Wanted

Upload goes flawlessly, and text file can be verified as uncorrupted when examining it through SSH on the server, need it to download intact.

[javascripts.corrupted.js.txt](https://github.com/proftpd/proftpd/files/1972805/javascripts.corrupted.js.txt)
[javascripts.functioning.js.txt](https://github.com/proftpd/proftpd/files/1972806/javascripts.functioning.js.txt)
[proftpd.conf.included.txt](https://github.com/proftpd/proftpd/files/1972807/proftpd.conf.included.txt)
[proftpd.conf.txt](https://github.com/proftpd/proftpd/files/1972808/proftpd.conf.txt)

[Fetch-FTP-Transcript.txt](https://github.com/proftpd/proftpd/files/1972817/Fetch-FTP-Transcript.txt)

### ProFTPD Version and Configuration

Plesk Onyx 17.8.11#6 on CentOS Linux 7.4.1708 (Core)‬ with:
```
psa-proftpd | 1.3.6-cos7.build1708180220.17

Compile-time Settings:
  Version: 1.3.6 (stable)
  Platform: LINUX [Linux 3.10.0-693.21.1.el7.x86_64 x86_64]
  Built: Tue Feb 20 2018 17:24:39 +07
  Built With:
    configure  '--with-modules=mod_ratio:mod_readme:mod_quotatab:mod_quotatab_file:mod_tls' '--enable-nls' '--enable-auth-pam' '--enable-ncurses' '--enable-ipv6' '--enable-buffer-size=8192' '--prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--libexecdir=/usr/lib64' '--datadir=/usr/share' '--sysconfdir=/etc' '--sharedstatedir=/usr/com' '--localstatedir=/var' '--libdir=/usr/lib64' '--includedir=/usr/include' '--infodir=/usr/share/info' '--mandir=/usr/share/man' '--build=x86_64-redhat-linux-gnu' '--host=x86_64-redhat-linux-gnu' 'build_alias=x86_64-redhat-linux-gnu' 'host_alias=x86_64-redhat-linux-gnu' 'CC=gcc' 'CFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic' 'LDFLAGS=' 'CPPFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic' 'CXX=g++' 'CXXFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic'

  CFLAGS: -g2 -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -Wall -fno-omit-frame-pointer
  LDFLAGS: -L$(top_srcdir)/lib  -rdynamic 
  LIBS:  -lssl -lcrypto -lcap  -lssl -lcrypto  -lpam -lsupp -lattr -lcrypt -ldl 

  Files:
    Configuration File:
      /etc/proftpd.conf
    Pid File:
      /var/proftpd.pid
    Scoreboard File:
      /var/proftpd.scoreboard

  Info:
    + Max supported UID: 4294967295
    + Max supported GID: 4294967295

  Features:
    - Autoshadow support
    - Controls support
    + curses support
    - Developer support
    - DSO support
    + IPv6 support
    + Largefile support
    - Lastlog support
    - Memcache support
    + ncursesw support
    + NLS support
    - Redis support
    - Sodium support
    + OpenSSL support (FIPS enabled)
    - PCRE support
    - POSIX ACL support
    + Shadow file support
    + Sendfile support
    + Trace support
    + xattr support

  Tunable Options:
    PR_TUNABLE_BUFFER_SIZE = 8192
    PR_TUNABLE_DEFAULT_RCVBUFSZ = 8192
    PR_TUNABLE_DEFAULT_SNDBUFSZ = 8192
    PR_TUNABLE_ENV_MAX = 2048
    PR_TUNABLE_GLOBBING_MAX_MATCHES = 100000
    PR_TUNABLE_GLOBBING_MAX_RECURSION = 8
    PR_TUNABLE_HASH_TABLE_SIZE = 40
    PR_TUNABLE_LOGIN_MAX = 256
    PR_TUNABLE_NEW_POOL_SIZE = 512
    PR_TUNABLE_PATH_MAX = 4096
    PR_TUNABLE_SCOREBOARD_BUFFER_SIZE = 80
    PR_TUNABLE_SCOREBOARD_SCRUB_TIMER = 30
    PR_TUNABLE_SELECT_TIMEOUT = 30
    PR_TUNABLE_TIMEOUTIDENT = 10
    PR_TUNABLE_TIMEOUTIDLE = 600
    PR_TUNABLE_TIMEOUTLINGER = 10
    PR_TUNABLE_TIMEOUTLOGIN = 300
    PR_TUNABLE_TIMEOUTNOXFER = 300
    PR_TUNABLE_TIMEOUTSTALLED = 3600
    PR_TUNABLE_XFER_SCOREBOARD_UPDATES = 10
```

**Comment by twouters** [comment]
Same issue has been reported on bugzilla: http://bugs.proftpd.org/show_bug.cgi?id=4352

**Comment by Castaglia** [comment]
The fixes for this (see https://github.com/proftpd/proftpd/pull/747) have been merged to master, and backported to the 1.3.6 branch.
